spec/requests/providers/means/add_other_state_benefits_controller_spec.rb

```diff
@@ -0,0 +1,126 @@
+require "rails_helper"
+
+RSpec.describe Providers::Means::AddOtherStateBenefitsController do
+  let(:legal_aid_application) { create(:legal_aid_application, :with_applicant) }
+  let(:setup) { login_as legal_aid_application.provider }
+  let(:state_benefit_transaction_type) { create(:transaction_type, :benefits) }
+
+  before do
+    state_benefit_transaction_type
+    setup
+    make_request
+  end
+
+  describe "GET /providers/applications/:legal_aid_application_id/means/add_other_state_benefits" do
+    subject(:make_request) { get providers_legal_aid_application_means_add_other_state_benefits_path(legal_aid_application) }
+
+    context "when the provider is not authenticated" do
+      let(:setup) { nil }
+
+      it_behaves_like "a provider not authenticated"
+    end
+
+    context "when the provider is authenticated" do
+      context "and no benefits have been added" do
+        it "returns an the expected page with the correct heading" do
+          expect(response).to have_http_status(:ok)
+          expect(response.body).to include("You have added 0 benefits")
+          expect(response.body).to include("Does your client receive any other benefits?")
```

Comment:
I'm not 100% sure about this as I haven't actually checked..... but I am pretty sure that for the dependants section if there are no dependants it shows the page which just says 'does your client have any dependants' and not 'you have added zero dependants' which seems a bit weird? Probably not a major issue but might be good to check that it is the same pattern for both pages in any case

Reply:
When routing through the flow, this does the same.  It was a test of calling the url directly and that it can handle being shown with and without records.  In practice, you are right, this should not be seen with zero records :+1: